op-supervisor/supervisor/backend/db/logs/db.go

```
@@ -193,8 +199,10 @@ func (db *DB) OpenBlock(blockNum uint64) (ref eth.BlockRef, logCount uint32, exe
 	db.rwLock.RLock()
 	defer db.rwLock.RUnlock()
 
+	// TODO: need to fix assumption that the first block is block 0
+	// This is only true for genesis-Interop chains.
 	if blockNum == 0 {
-		seal, err := db.StartingBlock()
+		seal, err := db.FirstSealedBlock()
```

`newIteratorAt` below handles the starting block case for non-genesis case. This `== 0` check is just to prevent the underflow.

---

That makes sense. But shouldn't we still change below logic to return an error `if seal.Number != 0`? Right now we just warn-log and return the block with a number > 0, which is not a defined behavior of `OpenBlock`. If this behavior is wanted from `OpenBlock`, we should document it. Otherwise let's return an error here. Maybe even a new error type like `ErrPreInterop` or `ErrPreActivation` to denote situations like this (the opposite of `ErrFuture`).

---

Fixed. And using the `types.ErrSkipped` error, which is meant for the kind of access to older data that is not indexed.